### PR TITLE
Add checkAjax support in Controller::redirect

### PR DESCRIPTION
**Pull request by ivanovyordan**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | yes |
| Fixed issues | #11058 |

### CODE REVIEW

File: framework/web/Controller.php [hunk 1]
```diff
@@ -139,11 +139,17 @@ public function beforeAction($action)
      * @param integer $statusCode the HTTP status code. Defaults to 302.
      * See <http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html>
      * for details about HTTP status code
+     *
+     * @param boolean $checkAjax whether to specially handle AJAX (and PJAX) requests. Defaults to true,
+     * meaning if the current request is an AJAX or PJAX request, then calling this method will cause the browser
+     * to redirect to the given URL. If this is false, a `Location` header will be sent, which when received as
```
Comment (SilverFire) [inline framework/web/Controller.php]: `false` -> <code>\`false\`</code>

**Review by SilverFire — CHANGES_REQUESTED** [review]
CHANGELOG and UPGRADE.md should be updated

File: framework/web/Controller.php (hunk 1 above)
Comment (ivanovyordan) [inline framework/web/Controller.php]: Setting default value to `false` would break BC right?

Comment (SilverFire) [inline framework/web/Controller.php]: Yeah

**Comment by rob006** [comment]
This breaks BC anyway, since it changes method signature.

**Comment by samdark** [comment]
Agree. It's a BC break.

File: framework/web/Controller.php (hunk 1 above)
Comment (ivanovyordan) [inline framework/web/Controller.php]: @SilverFire I made the changes.

Thanks for the feedback guys.

**Comment by SilverFire** [comment]
I'm fine with merging into 2.1 right now

**Comment by samdark** [comment]
Merged into 2.1: https://github.com/yiisoft/yii2/commit/0d72e37f3ce759a7286423eb32f19814bc2e0392

**Pull request closed without merging**